### PR TITLE
Fix profile updates based on profileId and release v8.0.2

### PR DESCRIPTION
**Pull request by Sassiest01**

### CODE REVIEW

File: entegywrapper/schemas/profile.py [hunk 1]
```diff
@@ -114,6 +114,7 @@ class ProfileCreate(BaseModel):
 
 
 class ProfileUpdate(BaseModel):
+    profileId: Optional[str] = None
```
Comment (willsawyerrrr) [inline entegywrapper/schemas/profile.py:117]: What problem does this fix? This seems like it enables us to update the profile ID which I don't think is true.

Comment (willsawyerrrr) [inline entegywrapper/schemas/profile.py:117]: What problem does this fix? This seems like it enables us to update the profile ID which I don't think is true.

Comment (Sassiest01) [reply entegywrapper/schemas/profile.py:117]: It doesn't update the profileId field, it allows us to use that field as the identifier when updating a profile. The same way we would send the external reference with a profile when updating based on that field.

Comment (willsawyerrrr) [reply entegywrapper/schemas/profile.py:117]: We can already identify a profile by profile ID (see [`EntegyAPI.update_profile`](https://github.com/SituDevelopment/entegy-sdk-python/blob/main/entegywrapper/profiles/profiles.py#L257-L265)).

You don't include the identifier in the profile object - it's a separate attribute in the body (see [docs](https://situ.entegysuite.com.au/Docs/Api/profile-update)).

Comment (Sassiest01) [reply entegywrapper/schemas/profile.py:117]: Ahh, I need to make another update, this is for the `sync_profiles` endpoint which has incorrect typing hinting for the profiles, it currently only accepts `Profile` where it should also accept the `ProfileUpdate` type.

Comment (Sassiest01) [reply entegywrapper/schemas/profile.py:117]: When syncing profiles, the profile needs to include the profile identifier, so if you are syncing based on the profileId, it should include that field.

Comment (willsawyerrrr) [reply entegywrapper/schemas/profile.py:117]: Ah, gotcha. Good catch!

File: entegywrapper/profiles/profiles.py [hunk 2]
```diff
@@ -330,7 +330,7 @@ def update_profile(
 def sync_profiles(
     self: EntegyAPI,
     update_reference_type: ProfileIdentifier,
-    profiles: list[Profile],
+    profiles: list[Profile | ProfileUpdate],
```
Comment (willsawyerrrr) [inline entegywrapper/profiles/profiles.py:333]: I know it pre-existed this PR, but maybe change `Profile` to `ProfileCreate`:

```suggestion
    profiles: list[ProfileCreate | ProfileUpdate],
```

You shouldn't be able to give this method a full `Profile` object.

Comment (Sassiest01) [reply entegywrapper/profiles/profiles.py:333]: It looks like `ProfileCreate` is very wrong, will need to update it

Comment (willsawyerrrr) [reply entegywrapper/profiles/profiles.py:333]: Oh shit, it is *very* wrong.

**Review by willsawyerrrr — APPROVED** [review]

**Pull request merged**